index.html

```diff
@@ -3737,55 +3739,50 @@ <h2>DID Resolution</h2>
-functions in addition to the <code>did</code> itself. Properties defined by this
-specification are in <a href="#did-resolution-input-metadata-properties"></a>.
-This input is REQUIRED, but the structure MAY be empty.
+A <a href="#metadata-structure">metadata structure</a> containing properties
+defined in <a href="#did-resolution-options"></a>. This input is
+REQUIRED, but the structure MAY be empty.
             </dd>
         </dl>
 
         <p>
-The output variables of these functions are as follows:
+The return value of <code>resolve</code> is the
+<a>didResolutionMetadata</a>, <a>didDocument</a>, and
+<a>didDocumentMetadata</a> properties. The return value of
+<code>resolveRepresentation</code> is the
+<a>didResolutionMetadata</a>, <a>didDocumentStream</a>, and
+<a>didDocumentMetadata</a> properties. These properties are described below:
         </p>
 
         <dl>
             <dt>
-did-resolution-metadata
+<dfn>didResolutionMetadata</dfn>
             </dt>
             <dd>
-              <p>
 A <a href="#metadata-structure">metadata structure</a> consisting of values
-relating to the results of the <a>DID resolution</a> process. This structure
-is REQUIRED and MUST NOT be empty.
-              </p>
-              <p>
-This metadata typically changes between invocations of the
-<code>resolve</code> and <code>resolveRepresentation</code> functions as it
-represents data about the resolution process itself. Properties defined by
-this specification are in <a href="#did-resolution-metadata-properties"></a>.
-              </p>
-              <p>
-If the resolution is successful, and if the <code>resolveRepresentation</code>
-function was called, this structure MUST contain a <code>contentType</code>
-property containing the mime-type of the <code>did-document-stream</code> in
-this result. If the resolution is not successful, this structure MUST contain
-an <code>error</code> property describing the error.
-              </p>
+relating to the results of the <a>DID resolution</a> process. This structure is
+REQUIRED and MUST NOT be empty. This metadata is defined by
+<a href="#did-resolution-metadata"></a>. If the resolution is successful this
+structure MUST contain a <code>contentType</code> property containing the
+mime-type of the <code>didDocument</code> in this result. If the resolution is
+not successful, this structure MUST contain an <code>error</code> property
+describing the error.
             </dd>
             <dt>
-did-document
+<dfn>didDocument</dfn>
             </dt>
             <dd>
 If the resolution is successful, and if the <code>resolve</code> function was
-called, this MUST be a <a>conforming DID document</a>. If the resolution is
+called, this MUST be a <a>DID document</a> (abstract data model) as described in
+<a href="#data-model"></a> that is capable of being transformed
+into a <a>conforming DID Document</a> (representation). If the resolution is
```

Comment:
```suggestion
into a <a>conforming DID Document</a> (representation), using the production rules specified by the representation. If the resolution is
```

+1 to your change, I like it a lot and think it helps with testability. Here is just a small additional suggestion to explicitly reference "production".